include/oneapi/dpl/internal/distributed_ranges_impl/shp/algorithms/for_each.hpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <sycl/sycl.hpp>
+
+#include <oneapi/dpl/internal/distributed_ranges_impl/detail/sycl_utils.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/algorithms/execution_policy.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/detail.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/init.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/util.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/zip_view.hpp>
+
+namespace oneapi::dpl::experimental::dr::shp
+{
+
+template <typename ExecutionPolicy, distributed_range R, typename Fn>
+void
+for_each(ExecutionPolicy&& policy, R&& r, Fn&& fn)
```

Comment:
Similar comment about rvalue reference of `fn`. The same goes for the following algorithms.

Reply:
same answer :)

Reply:
ok, applied